lambdas/id_sync/src/record_processor.py

```diff
@@ -1,115 +1,98 @@
+    logger.info("Processing record with SQS messageId: %s", event_record.get("messageId"))
     body_text = event_record.get("body", "")
 
-    # convert body to json (try JSON first, then fall back to Python literal)
-    if isinstance(body_text, str):
-        try:
-            body = json.loads(body_text)
-        except json.JSONDecodeError:
-            try:
-                body = ast.literal_eval(body_text)
-            except (ValueError, SyntaxError):
-                logger.error("Failed to parse body: %s", body_text)
-                return {"status": "error", "message": "Invalid body format"}
-    else:
-        body = body_text
+    try:
+        body = json.loads(body_text)
+    except json.JSONDecodeError:
+        logger.error("Failed to parse body: %s", body_text)
+        return {"status": "error", "message": "Invalid body format"}
 
     nhs_number = body.get("subject")
-    logger.info("process record NHS number: %s", nhs_number)
-    if nhs_number:
-        return process_nhs_number(nhs_number)
+    logger.info("Processing MNS event with id: %s", body.get("id"))
+
+    if not nhs_number:
+        logger.info("No NHS number found in event record")
+        return {"status": "error", "message": "No NHS number found in event record"}
 
-    logger.info("No NHS number found in event record")
-    return {"status": "error", "message": "No NHS number found in event record"}
+    return process_nhs_number(nhs_number)
 
 
 def process_nhs_number(nhs_number: str) -> Dict[str, Any]:
-    # get patient details from PDS
-    new_nhs_number = pds_get_patient_id(nhs_number)
+    try:
+        pds_patient_resource = pds_get_patient_details(nhs_number)
+    except IdSyncException as e:
+        return make_status(str(e), status="error")
+
+    if not pds_patient_resource:
+        return make_status("No patient ID found for NHS number")
 
-    if not new_nhs_number:
-        return make_status("No patient ID found for NHS number", nhs_number)
+    new_nhs_number = get_nhs_number_from_pds_resource(pds_patient_resource)
 
     if new_nhs_number == nhs_number:
-        return make_status("No update required", nhs_number)
+        return make_status("No update required")
 
-    logger.info("Update patient ID from %s to %s", nhs_number, new_nhs_number)
+    logger.info("NHS Number has changed. Performing updates on relevant IEDS records")
 
     try:
-        # Fetch PDS Patient resource and IEDS resources for the old NHS number
-        pds_patient_resource, ieds_resources = fetch_pds_and_ieds_resources(nhs_number)
+        # Fetch the IEDS resources for the old NHS number
+        ieds_resources = fetch_ieds_resources(nhs_number)
     except Exception as e:
-        logger.exception("process_nhs_number: failed to fetch demographic details: %s", e)
-        return make_status(str(e), nhs_number, "error")
+        logger.exception("process_nhs_number: failed to fetch ieds resources: %s", e)
+        return make_status(str(e), status="error")
```

Comment:
`"error"` should be a constant. Although given we only ever use `"error"` or `"success"`, the status should probably just be a `success` boolean